### PR TITLE
Transpose profiler fix

### PR DESCRIPTION
**Pull request by arai713**

### CODE REVIEW

File: profiler/include/profiler/profile_transpose_impl.hpp [hunk 1]
```diff
@@ -153,10 +153,6 @@ bool profile_transpose_impl(int do_verification,
             std::cout << "Perf: " << std::setw(10) << ave_time << " ms, " << tflops << " TFlops, "
                       << gb_per_sec << " GB/s, " << op_name << std::endl;
 
-            // pass = pass & ck::utils::check_err(b_device_result, b_host_result);
```
Comment (bartekxk) [inline profiler/include/profiler/profile_transpose_impl.hpp:156]: How result is verified?

File: profiler/src/profile_transpose.cpp [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+};
+
+#define OP_NAME "transpose"
+#define OP_DESC "Transpose"
+
+struct TransposeArgParser
+{
+    std::unordered_map<std::string, std::vector<int>> long_opts = {{"lengths", {}}};
+
+    bool parse_opt(int argc, char* argv[], const std::string& key, int i)
```
Comment (bartekxk) [inline profiler/src/profile_transpose.cpp]: Please add const where it is possible (`argc`, `argv`, `pos`)

File: profiler/src/profile_transpose.cpp [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+            }
+        }
+    }
+};
+
+int profile_transpose(int argc, char* argv[])
+{
+    if(argc != 7)
+    {
+        printf("arg1: tensor operation (" OP_NAME ": " OP_DESC ")\n");
```
Comment (bartekxk) [inline profiler/src/profile_transpose.cpp]: Could You move this to static `void print_helper_msg()`?

File: profiler/src/profile_transpose.cpp [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+    TransposeArgParser arg_parser;
+
+    const auto data_type       = static_cast<DataType>(std::stoi(argv[2]));
+    const bool do_verification = std::stoi(argv[3]);
+    const int init_method      = std::stoi(argv[4]);
+    const bool do_log          = std::stoi(argv[5]);
+    const bool time_kernel     = std::stoi(argv[6]);
+    arg_parser(argc, argv);
+    const std::vector<ck::index_t> lengths = arg_parser.long_opts["lengths"];
+    // std::vector<ck::index_t> lengths = std::stoi(argv[7]);
```
Comment (bartekxk) [inline profiler/src/profile_transpose.cpp]: Please remove

File: profiler/src/profile_transpose.cpp [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+
+    using F32 = float;
+    using F16 = ck::half_t;
+
+    auto profile = [&](auto a_type, auto b_type) {
+        using ADataType = decltype(a_type);
+        using BDataType = decltype(b_type);
+
+        bool pass = ck::profiler::profile_transpose_impl<ADataType, BDataType, 5>(
```
Comment (bartekxk) [inline profiler/src/profile_transpose.cpp]:
```suggestion
        constexpr ck::index_t NumDim = 5;

        bool pass = ck::profiler::profile_transpose_impl<ADataType, BDataType, NumDim>(

```

File: profiler/include/profiler/profile_transpose_impl.hpp (hunk 1 above)
Comment (arai713) [reply profiler/include/profiler/profile_transpose_impl.hpp:156]: The result is verified in line 127

Comment (bartekxk) [reply profiler/include/profiler/profile_transpose_impl.hpp:156]: You're right. Just to make sure. Why kernel is run twice? Kernel uses some atomic adds so the first one is for check the result and second one for measure the performance? If yes, could You put some comment?

File: profiler/src/profile_transpose.cpp [hunk 6]
```diff
@@ -0,0 +1,112 @@
+// SPDX-License-Identifier: MIT
+// Copyright (c) 2018-2023, Advanced Micro Devices, Inc. All rights reserved.
```
Comment (bartekxk) [inline profiler/src/profile_transpose.cpp]:
```suggestion
// Copyright (c) 2024, Advanced Micro Devices, Inc. All rights reserved.

```

**Comment by bartekxk** [comment]
Please update copyrights in edited files.

**Review by bartekxk — APPROVED** [review]

**Comment by bartekxk** [comment]
LGTM, I discovered that there are missing copyrigts in elementwise_permute examples. If You prefer You can fix this in this PR, if not, I prepared PR https://github.com/ROCm/composable_kernel/pull/1118

**Pull request merged**